<launch>
    <!-- <arg name="input_topic" default="/velodyne_points" /> -->
    <arg name="input_topic" default="/source" />    
    <arg name="no_ground_point_topic" default="/points_no_ground"/>
    <arg name="ground_point_topic" default="/points_ground"/>
    <arg name="all_points_topic" default="/all_points"/>

    <!-- clip the points above sensor_height+clip_height -->
    <arg name="clip_height" default="4.0"/>
    <!-- the height of lidar position  -->
    <!-- <arg name="sensor_height" default="1.77"/> -->
    <arg name="sensor_height" default="1.73"/>    
    <!-- the min distance to be remove -->
    <!-- <arg name="min_distance" default="2.0"/> -->
    <arg name="min_distance" default="0.0"/>    
    <!-- the max distance to be remove -->
    <!-- <arg name="max_distance" default="75.0"/> -->
    <arg name="max_distance" default="1000.0"/>    
    <!-- we use 32 lidar -->
    <!-- <arg name="sensor_model" default="32"/> -->
    <arg name="sensor_model" default="64"/>    
    <!-- Num of Iteration -->
    <arg name="num_iter" default="3"/>
    <!-- Num of LPR -->
    <arg name="num_lpr" default="20"/>
    <!-- Seeds Threshold  default 1.2 -->
    <!-- <arg name="th_seeds" default="1.8"/>     -->
    <arg name="th_seeds" default="1.2"/>
    <!-- Distance Threshold   default 0.3-->
    <arg name="th_dist" default="0.3"/>

    <arg name="pcd_read1" default="true" />
    <group if="$(arg pcd_read1)">
        <param name="pcd_path" value="/home/meng/subject/data_preprocess/kitti_pre_processed/sequence00/1151_1200_view/source.pcd" />             
        <node pkg="plane_ground_filter" name="read_pub_pcd" type="read_pub_pcd.py" output="screen">
        </node> 
    </group>

    <node pkg="plane_ground_filter" type="plane_ground_filter_node" name="plane_ground_filter_node" output="screen">
        <param name="input_topic" value="$(arg input_topic)"/>
        <param name="no_ground_point_topic" value="$(arg no_ground_point_topic)"/>
        <param name="ground_point_topic" value="$(arg ground_point_topic)"/>
        <param name="all_points_topic" value="$(arg all_points_topic)"/>

        <param name="sensor_height" value="$(arg sensor_height)" />
        <param name="clip_height" value="$(arg clip_height)" />
        <param name="min_distance" value="$(arg min_distance)" />
        <param name="max_distance" value="$(arg max_distance)" />
        <param name="sensor_model" value="$(arg sensor_model)" />
        <param name="num_iter" value="$(arg num_iter)" />
        <param name="num_lpr" value="$(arg num_lpr)" />
        <param name="th_seeds" value="$(arg th_seeds)" />
        <param name="th_dist" value="$(arg th_dist)" />
    </node>

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz_2" args="-d $(find plane_ground_filter)/2.rviz" />
    </group>
</launch>